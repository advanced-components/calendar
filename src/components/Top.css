.calendar .top {
  height: 40px;
  width: 100%;
  border-bottom: 1px solid #eee;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 0 15px;
}

.calendar .top .add, .calendar .top .view-change {
  display: flex;
  align-items: center;
}