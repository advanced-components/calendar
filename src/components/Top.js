import React from 'react';
import Navigator from './Navigator';
import './Top.css';

export const Top = () => (
  <div className="top">
    <div className="top__left">
      <button type="button">+</button>
    </div>
    <div className="top__middle">
      <Navigator />
    </div>
    <div className="top__right">
      <button type="button">DAY</button>
      <button type="button">WEEK</button>
      <button type="button">MONTH</button>
    </div>
  </div>
);

export default Top;
