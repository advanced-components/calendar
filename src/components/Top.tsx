import React from 'react';
import './Top.css';

import {
  NAVIGATE_TODAY,
  NAVIGATE_NEXT,
  NAVIGATE_PREVIOUS,
  DAY_VIEW,
  WEEK_VIEW,
  MONTH_VIEW
} from '../constants';

export const Top: React.FunctionComponent<{
  onShowModal: Function;
  onNavigate: Function;
  onChangeView: Function;
}> = ({ onShowModal, onNavigate, onChangeView }) => (
  <div className='top'>
    <div className='top__left'>
      <button
        className='add-event-button'
        type='button'
        onClick={() => onShowModal()}
      >
        +
      </button>
    </div>
    <div className='top__middle'>
      <div className='navigation'>
        <button
          className='navigation-button navigation-button__today'
          type='button'
          onClick={() => onNavigate(NAVIGATE_TODAY)}
        >
          TODAY
        </button>
        <button
          className='navigation-button navigation-button__next'
          type='button'
          onClick={() => onNavigate(NAVIGATE_NEXT)}
        >
          {'<'}
        </button>
        <button
          className='navigation-button navigation-button__previous'
          type='button'
          onClick={() => onNavigate(NAVIGATE_PREVIOUS)}
        >
          {'>'}
        </button>
      </div>
    </div>
    <div className='top__right'>
      <button
        className='change-view-button change-view-button__day'
        onClick={() => onChangeView(DAY_VIEW)}
        type='button'
      >
        DAY
      </button>
      <button
        className='change-view-button change-view-button__week'
        onClick={() => onChangeView(WEEK_VIEW)}
        type='button'
      >
        WEEK
      </button>
      <button
        className='change-view-button change-view-button__month'
        onClick={() => onChangeView(MONTH_VIEW)}
        type='button'
      >
        MONTH
      </button>
    </div>
  </div>
);

export default Top;
