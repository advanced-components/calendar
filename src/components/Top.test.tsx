import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import {
  NAVIGATE_TODAY,
  NAVIGATE_NEXT,
  NAVIGATE_PREVIOUS,
  DAY_VIEW,
  WEEK_VIEW,
  MONTH_VIEW
} from '../constants';

import { Top } from './Top';

const onShowModal = jest.fn();
const onNavigate = jest.fn();
const onChangeView = jest.fn();

const defaultProps = {
  onShowModal: onShowModal,
  onNavigate: onNavigate,
  onChangeView: onChangeView
};

describe('<Top />', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Top {...defaultProps} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('call onShowModal()', () => {
    const component = shallow(<Top {...defaultProps} />);
    component.find('.add-event-button').simulate('click');
    expect(onShowModal).toHaveBeenCalled();
  });

  it('call onNavigate(NAVIGATE_TODAY)', () => {
    const component = shallow(<Top {...defaultProps} />);
    component.find('.navigation-button__today').simulate('click');
    expect(onNavigate).toHaveBeenCalledWith(NAVIGATE_TODAY);
  });

  it('call onNavigate(NAVIGATE_NEXT)', () => {
    const component = shallow(<Top {...defaultProps} />);
    component.find('.navigation-button__next').simulate('click');
    expect(onNavigate).toHaveBeenCalledWith(NAVIGATE_NEXT);
  });

  it('call onNavigate(NAVIGATE_PREVIOUS)', () => {
    const component = shallow(<Top {...defaultProps} />);
    component.find('.navigation-button__previous').simulate('click');
    expect(onNavigate).toHaveBeenCalledWith(NAVIGATE_PREVIOUS);
  });

  it('call onChangeView(DAY_VIEW)', () => {
    const component = shallow(<Top {...defaultProps} />);
    component.find('.change-view-button__day').simulate('click');
    expect(onChangeView).toHaveBeenCalledWith(DAY_VIEW);
  });

  it('call onChangeView(WEEK_VIEW)', () => {
    const component = shallow(<Top {...defaultProps} />);
    component.find('.change-view-button__week').simulate('click');
    expect(onChangeView).toHaveBeenCalledWith(WEEK_VIEW);
  });

  it('call onChangeView(MONTH_VIEW)', () => {
    const component = shallow(<Top {...defaultProps} />);
    component.find('.change-view-button__month').simulate('click');
    expect(onChangeView).toHaveBeenCalledWith(MONTH_VIEW);
  });
});
