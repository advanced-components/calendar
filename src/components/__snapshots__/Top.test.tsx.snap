// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Top /> renders correctly 1`] = `
<div
  className="top"
>
  <div
    className="top__left"
  >
    <button
      className="add-event-button"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="top__middle"
  >
    <div
      className="navigation"
    >
      <button
        className="navigation-button navigation-button__today"
        onClick={[Function]}
        type="button"
      >
        TODAY
      </button>
      <button
        className="navigation-button navigation-button__next"
        onClick={[Function]}
        type="button"
      >
        &lt;
      </button>
      <button
        className="navigation-button navigation-button__previous"
        onClick={[Function]}
        type="button"
      >
        &gt;
      </button>
    </div>
  </div>
  <div
    className="top__right"
  >
    <button
      className="change-view-button change-view-button__day"
      onClick={[Function]}
      type="button"
    >
      DAY
    </button>
    <button
      className="change-view-button change-view-button__week"
      onClick={[Function]}
      type="button"
    >
      WEEK
    </button>
    <button
      className="change-view-button change-view-button__month"
      onClick={[Function]}
      type="button"
    >
      MONTH
    </button>
  </div>
</div>
`;
